sdk/src/wallet/operations/transaction/high_level/delegation/delay.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::output::{
+        AddressUnlockCondition, BasicOutput, DelegationId, DelegationOutputBuilder, MinimumOutputAmount,
+    },
+    wallet::{types::TransactionWithMetadata, Wallet},
+};
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    /// Delay a delegation's claiming. The `reclaim_excess` flag indicates whether excess value over the minimum storage
+    /// requirements will be moved to a basic output that is unlockable by the same address which controls the
+    /// delegation. Otherwise it will be added to a new delegation. In this manner, one can delegate for one epoch
+    /// at a time and never lose out on any rewards.
+    pub async fn delay_delegation_claiming(
+        &self,
+        delegation_id: DelegationId,
+        reclaim_excess: bool,
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let prepared_transaction = self
+            .prepare_delay_delegation_claiming(delegation_id, reclaim_excess)
+            .await?;
+
+        self.sign_and_submit_transaction(prepared_transaction, None, None).await
+    }
+
+    /// Prepare to delay a delegation's claiming. The `reclaim_excess` flag indicates whether excess value
+    /// over the minimum storage requirements will be moved to a basic output that is unlockable by the same address
+    /// which controls the delegation.
+    /// Otherwise it will be added to a new delegation. In this manner, one can delegate for one epoch at a time and
+    /// never lose out on any rewards.
+    pub async fn prepare_delay_delegation_claiming(
+        &self,
+        delegation_id: DelegationId,
+        reclaim_excess: bool,
+    ) -> crate::wallet::Result<PreparedTransactionData> {
+        let delegation_output = self
+            .data()
+            .await
+            .unspent_delegation_output(&delegation_id)
+            .ok_or(crate::wallet::Error::MissingDelegation(delegation_id))?
+            .output
+            .clone();
+        let protocol_parameters = self.client().get_protocol_parameters().await?;
+
+        let min_delegation_amount = delegation_output.minimum_amount(protocol_parameters.storage_score_parameters());
+
+        // TODO: Should we return an error if `reclaim_excess == true` and `can_reclaim == false`?
```

Comment:
I think that makes sense to return an error if that doesn't work out, because if one provides `reclaim_excess == true` one probably expect that it delegates all the time
And if we notice that this error gets annoying because it happens often and we would actually be fine without, we an also remove again

Reply:
okie dokie

Reply:
I ended up refactoring so that this error is not possible